v3/src/execution/__tests__/execution.test.ts

```diff
@@ -0,0 +1,436 @@
+import {
+  parseGraphqlRequest,
+  validateGraphqlRequest,
+  executeGraphqlRequest,
+  processGraphqlRequest,
+} from '..';
+import { astSerializer } from '../../snapshotSerializers';
+import { VariableValues, gql } from 'apollo-server-core';
+import { buildServiceDefinition } from '@apollographql/apollo-tools';
+import { GraphQLError } from 'graphql';
+
+expect.addSnapshotSerializer(astSerializer);
+
+// This is a handy trick for syntax highlighting GraphQL query strings in editors
+// like VS Code that depend on the gql`` or graphql`` tagged template literals.
+const graphql = String.raw;
+
+// An array of tuples, each tuple representing a test.
+// A single tuple has the shape: [testName, { ...testParams }]
+type JestInput<T> = [string, T][];
```

Comment:
I really like this! It took me a second to figure it out, but I really like it.

One thought: what would you think about making these objects? like:

```ts
type JestInput<T> = { [test: string]: T }
validQueries: JestInput<{query: string, operationName?: string, variables?: VariableValues}> = {
  [`basic — { people { name } }`]: {
    query: graphql `...`,
  }
}
```

Reply:
(Mostly to indicate uniqueness and have the linter catch "oops, I copy-pasted this test case but forgot to rename it"?)

Reply:
Yes! Done!